fetch-in-react/src/app.jsx

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import UserList from './user-list';
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      users: [],
+      isLoading: true
+    };
+  }
+
+  componentDidMount() {
+    fetch('https://jsonplaceholder.typicode.com/users')
+      .then(res => res.json())
+      .then(data => {
+        const user = data;
+        this.setState({ users: user });
+      })
+      .then(this.setState({ isLoading: false }));
```

Comment:
This setState on line 20 is happening prematurely because `.then()` takes a function _definition_ as its argument.

```js
      .then(() => this.setState({ isLoading: false }));

```